src/audio/SDL_sysaudio.h

```diff
@@ -48,7 +48,17 @@
 #define DEFAULT_AUDIO_RECORDING_CHANNELS 1
 #define DEFAULT_AUDIO_RECORDING_FREQUENCY 44100
 
-#define AUDIO_SPECS_EQUAL(x, y) (((x).format == (y).format) && ((x).channels == (y).channels) && ((x).freq == (y).freq))
+SDL_FORCE_INLINE SDL_bool SDL_AudioSpecsEqual(const SDL_AudioSpec *a, const SDL_AudioSpec *b)
+{
+    if ((a->format != b->format) || (a->channels != b->channels) || (a->freq != b->freq) || (a->use_channel_map != b->use_channel_map)) {
+        return SDL_FALSE;
+    } else if (a->use_channel_map && (SDL_memcmp(a->channel_map, b->channel_map, sizeof (a->channel_map[0]) * a->channels) != 0)) {
+        return SDL_FALSE;
+    }
+    return SDL_TRUE;
+}
+
+#define AUDIO_SPECS_EQUAL(x, y) SDL_AudioSpecsEqual(&(x), &(y))
```

Comment:
Any reason we need the macro? Can we just call the function?

Reply:
Was just minimizing code changes, but I'll clean this out.

Reply:
Fixed in 5102dc74b34c85d381630b4b4915bf92180cf24b.